### DISCUSSION TITLE
Tooltip for other HTML elements

### DISCUSSION THREAD

**Discussion by akkie** [chakra-ui/zag#1924] [Ideas]
Hi,

it seems, that the toolbar component can only be used with button elements. At least from a typing perspective, because `api.getTriggerProps()` returns `HTMLButtonAttributes`. It would be helpful to use the toolbar with anchor elements as well. As example, in SvelteKit there are additional data attributes which can be used to preload data or code, when hovering over an anchor element. For my main menu as example I would like to use that functionality with an icon based menubar. But I need tooltip to show some more information about the menu item. 

Is there any reason, why the current implementation only supports buttons?

**Comment by segunadebayo** [comment]
This is largely a design limitation we have in Zag with regards the typings.

At the moment, we can’t express that an element should support multiple “types” (button or links for example). What we can do is to loosen the type to support any element but that’s not helpful either.

I recommend casting the types on your end for now.
